### PR TITLE
[Merged by Bors] - Add Test Timeout

### LINKED ISSUE #3279: ci: systest job should always run make clean
`make run` may timeout or get stuck, ci job needs a fail-safe approach to kill `make run` process after N minutes and run `make clean`.

### PR DESCRIPTION
**Pull request by web3rover**
## Motivation

Closes #3279

## TODO

- [ ] Explain motivation or link existing issue(s)
- [ ] Test changes and document test plan
- [ ] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by web3rover** [comment]
bors try

**Comment by web3rover** [comment]
bors try

**Comment by web3rover** [comment]
bors try

**Review by evt — APPROVED** [review]

**Comment by web3rover** [comment]
bors try

**Comment by countvonzero** [comment]
@narayanprusty does this PR also fixes https://github.com/spacemeshos/go-spacemesh/issues/3316?

**Comment by dshulyak** [comment]
@countvonzero clearly not. that issue is due to some internal k8s processes that we need to understand

**Comment by web3rover** [comment]
bors try

**Comment by noamnelke** [comment]
bors try-

**Comment by noamnelke** [comment]
bors try

**Comment by noamnelke** [comment]
bors try-

**Comment by noamnelke** [comment]
bors try

**Comment by noamnelke** [comment]
bors try-

**Comment by web3rover** [comment]
bors try

**Comment by web3rover** [comment]
bors try

**Comment by web3rover** [comment]
bors try

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -207,6 +207,7 @@ jobs:
       run: cd systest && make push
 
     - name: Run tests
+      timeout-minutes: 40
```
Comment (dshulyak) [inline .github/workflows/ci.yml]: default timeout for a test is set to 60m. this timeout needs to be atleast as large

**Review by dshulyak — APPROVED** [review]

**Comment by web3rover** [comment]
bors merge

**Pull request closed without merging**